gluten-flink/ut/src/test/java/org/apache/gluten/streaming/api/operators/GlutenStreamFilterTest.java

```diff
@@ -0,0 +1,368 @@
+    TypedExpr veloxFilterCondition =
+        RexNodeConverter.toTypedExpr(filterCondition, conversionContext);
+
+    io.github.zhztheplayer.velox4j.type.RowType veloxType =
+        (io.github.zhztheplayer.velox4j.type.RowType)
+            org.apache.gluten.util.LogicalTypeConverter.toVLType(flinkRowType);
+
+    PlanNode veloxPlan =
+        new FilterNode(
+            PlanNodeIdGenerator.newId(), List.of(new EmptyNode(veloxType)), veloxFilterCondition);
+
+    GlutenSingleInputOperator operator =
+        new TestGlutenSingleInputOperator(
+            new StatefulPlanNode(veloxPlan.getId(), veloxPlan),
+            PlanNodeIdGenerator.newId(),
+            veloxType,
+            Map.of(veloxPlan.getId(), veloxType));
+
+    TypeSerializer<RowData> serializer = typeInfo.createSerializer(new SerializerConfigImpl());
+    OneInputStreamOperatorTestHarness<RowData, RowData> harness =
+        new OneInputStreamOperatorTestHarness<>(operator, serializer);
+
+    harness.setup(typeInfo.createSerializer(new SerializerConfigImpl()));
+    harness.open();
+
+    long timestamp = 0L;
+    for (RowData row : testInputData) {
+      harness.processElement(new StreamRecord<>(row, timestamp++));
+    }
+
+    List<RowData> actualOutput = extractOutputFromHarness(harness);
+    assertRowDataListEquals(actualOutput, expectedOutput);
+
+    harness.close();
+  }
+
+  private  static class TestGlutenSingleInputOperator extends GlutenSingleInputOperator {
+    public TestGlutenSingleInputOperator(
+            StatefulPlanNode plan, String id,
+            io.github.zhztheplayer.velox4j.type.RowType inputType, Map<String, io.github.zhztheplayer.velox4j.type.RowType> outTypes) {
+      super(plan, id, inputType, outTypes);
+    }
+    @Override
+    public void processElement(StreamRecord<RowData> element) {
+      try {
+        super.processElement(element);
+      } catch (Exception e) {
+        if(e.getMessage() != null && (e.getMessage().contains("ResourceHandle not found") || e.getMessage().contains("Failed to reattach current thread to JVM"))) {
+
+        } else {
+          throw e;
+        }
+      }
+    }
+  }
+
+  private List<RowData> extractOutputFromHarness(
+      OneInputStreamOperatorTestHarness<RowData, RowData> harness) {
+    Queue<Object> outputQueue = harness.getOutput();
+    List<RowData> actualOutput = new ArrayList<>();
+    while (!outputQueue.isEmpty()) {
+      Object record = outputQueue.poll();
+      if (record instanceof StreamRecord) {
+        actualOutput.add(((StreamRecord<RowData>) record).getValue());
+      }
+    }
+    return actualOutput;
+  }
+
+  private void assertRowDataListEquals(List<RowData> actual, List<RowData> expected) {
+    assertThat(actual).hasSize(expected.size());
+    for (int i = 0; i < expected.size(); i++) {
+      assertRowDataEquals(actual.get(i), expected.get(i));
+    }
+  }
+
+  private void assertRowDataEquals(RowData actual, RowData expected) {
```

Comment:
naming suggestion: `checkEquals`

Please check Flink code to see whether it has a public compare method that we can directly use.
 
It would be better to make this function general (maybe, move it to a util class for other tests to use), not bound to this test class. For general case, we don't know each column's data type beforehand. Given any RowData, we can get each column's data type and using the corresponding getter method to get the data for comparison.

For strict comparison, please check whether they have same fields number obtained by `getArity()`.
Please also make sure null data is allowed in this check. It would be better to add some test cases with NULL data.